### PR TITLE
New config command

### PR DESCRIPTION
**Pull request by Innectic**
## Requirements

 - [x] Only PG-rated language used (code, commits, etc.)
 - [x] Descriptive commit messages
 - [x] Changes have been tested
 - [x] Changes do not break any existing functionalities

### CODE REVIEW

**Review by RPiAwesomeness — APPROVED** [review]
👌

File: cactusbot/commands/magic/config.py [hunk 1]
```diff
@@ -16,131 +18,289 @@
     })
 
 
-async def _update_spam_config(api, scope, field, value):
+async def _update_config(api, scope, field, value):
```
Comment (2Cubed) [inline cactusbot/commands/magic/config.py:21]: Could make this cleaner (and more scalable):

```python
In [1]: def _deep_dict(*fields):
   ...:     if len(fields) == 2:
   ...:         return {fields[0]: fields[1]}
   ...:     return {fields[0]: _deep_dict(*fields[1:])}
   ...: 

In [2]: def _update_config(api, *fields):
   ...:     return (api, _deep_dict(*fields))
   ...: 

In [3]: _update_config("API", "announce", "follow", "announce", False)
Out[3]: ('API', {'announce': {'follow': {'announce': False}}})

In [4]: _update_config("API", "spam", "allowUrls", True)
Out[4]: ('API', {'spam': {'allowUrls': True}})
```

File: cactusbot/commands/magic/config.py [hunk 2]
```diff
@@ -16,131 +18,289 @@
...
+        """Follow subcommand."""
 
-        @Command.command()
-        async def follow(self, value):
-            """Follow subcommand."""
+        @Command.command(role="moderator", name="follow")
+        async def default(self, value=""):
+            """Get status, and message of the follow event, or toggle."""
+
+            if not value:
```
Comment (2Cubed) [inline cactusbot/commands/magic/config.py:60]: This is annoying, but currently necessary. Would like to add some form of pattern matching to `command.py` at some point, but for now, this works.

**Comment by 2Cubed** [comment]
Started reviewing when it was still open... lol. Looks good, though - just wanted to comment on those things briefly. (Don't need to change anything, TBH.)

**Pull request merged**